pkg/resources/kafka/configmap.go

```diff
@@ -63,46 +63,36 @@ func (r *Reconciler) getConfigProperties(bConfig *v1beta1.BrokerConfig, id int32
-			log.Error(err, "setting ssl.truststore.password parameter in broker configuration resulted an error")
+		if err := config.Set("cruise.control.metrics.reporter.ssl.truststore.password", clientPass); err != nil {
+			log.Error(err, "setting cruise.control.metrics.reporter.ssl.truststore.password parameter in broker configuration resulted an error")
 		}
-		if err := config.Set("ssl.client.auth", "required"); err != nil {
-			log.Error(err, "setting ssl.client.auth parameter in broker configuration resulted an error")
+		if err := config.Set("cruise.control.metrics.reporter.ssl.keystore.location", clientKeystorePath+"/"+v1alpha1.TLSJKSKeyStore); err != nil {
+			log.Error(err, "setting cruise.control.metrics.reporter.ssl.keystore.location parameter in broker configuration resulted an error")
 		}
-
-		// Add Cruise Control SSL configuration
-		if util.IsSSLEnabledForInternalCommunication(r.KafkaCluster.Spec.ListenersConfig.InternalListeners) {
-			if err := config.Set("cruise.control.metrics.reporter.security.protocol", "SSL"); err != nil {
-				log.Error(err, "setting cruise.control.metrics.reporter.security.protocol in broker configuration resulted an error")
-			}
-			if err := config.Set("cruise.control.metrics.reporter.ssl.truststore.location", clientKeystorePath+"/"+v1alpha1.TLSJKSTrustStore); err != nil {
-				log.Error(err, "setting cruise.control.metrics.reporter.ssl.truststore.location in broker configuration resulted an error")
-			}
-			if err := config.Set("cruise.control.metrics.reporter.ssl.keystore.location", clientKeystorePath+"/"+v1alpha1.TLSJKSKeyStore); err != nil {
-				log.Error(err, "setting cruise.control.metrics.reporter.ssl.keystore.location parameter in broker configuration resulted an error")
-			}
-			if err := config.Set("cruise.control.metrics.reporter.ssl.keystore.password", clientPass); err != nil {
-				log.Error(err, "setting cruise.control.metrics.reporter.ssl.keystore.password parameter in broker configuration resulted an error")
-			}
+		if err := config.Set("cruise.control.metrics.reporter.ssl.keystore.password", clientPass); err != nil {
+			log.Error(err, "setting cruise.control.metrics.reporter.ssl.keystore.password parameter in broker configuration resulted an error")
 		}
 	}
-
+	if err := config.Set("password.encoder.secret", "secret"); err != nil {
+		log.Error(err, "setting cruise.control.metrics.reporter.ssl.keystore.password parameter in broker configuration resulted an error")
+	}
 	// Add Cruise Control Metrics Reporter configuration
 	if err := config.Set("metric.reporters", "com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter"); err != nil {
 		log.Error(err, "setting metric.reporters in broker configuration resulted an error")
 	}
-	brokerPort, err := kafkautils.GetBrokerContainerPort(r.KafkaCluster)
 	if err != nil {
 		log.Error(err, "getting Kafka broker port for Cruise Control failed")
 	}
```

Comment:
This is a leftover code. But I would add the `brokerPort, err := kafkautils.GetBrokerContainerPort(r.KafkaCluster)` before this.